File: src/model/DefaultDeliveryStop.java
package model;

import java.util.ArrayList;

public class DefaultDeliveryStop {
    
    private long id;
    private ArrayList<Customer> customers;
    private long geoLocID;
    
    public DefaultDeliveryStop(long id, long geoLocID) {
        this.id = id;
        this.geoLocID = geoLocID;
    }

    /**
     * @return the id
     */
    public long getID() {
        return id;
    }

    /**
     * @return the customers
     */
    public ArrayList<Customer> getCustomers() {
        return customers;
    }

    /**
     * @param customers the customers to set
     */
    public void setCustomers(ArrayList<Customer> customers) {
        this.customers = customers;
    }

    /**
     * @return the geoLoc
     */
    public long getGeoLocID() {
        return this.geoLocID;
    }
}

File: Model/DefaultDeliveryStop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace Model
{
    [DataContract()]
    //[KnownType(typeof(Customer))]
    public class DefaultDeliveryStop
    {
        private long _geoLocID;

        [DataMember()]
        public long ID { get; private set; }
        [DataMember()]
        public List<Customer> Customers { get; set; }
        [DataMember()]
        public int? SequenceNbr { get; private set; }
        [DataMember()]
        public GeoLoc GeoLoc { get; set; }
        [DataMember()]
        public long GeoLocID
        {
            get { return GeoLoc == null ? _geoLocID : GeoLoc.ID; }
            private set { _geoLocID = value; }
        }

        public DefaultDeliveryStop(long ID, long geoLocId)
        {
            this.ID = ID;
            GeoLocID = geoLocId;
        }

        public DefaultDeliveryStop(long ID, long geoLocId, int? SequenceNbr)
        {
            this.ID = ID;
            this._geoLocID = geoLocId;
            this.SequenceNbr = SequenceNbr;
        }
    }
}
